### PR TITLE
Direct deposit legacy cleanup

### PR DESCRIPTION
**Pull request by tpharrison**
## Summary
Remove disability compensations (old C&P direct deposit) code.  It has been replaced.

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/81264

## Testing done
Removed associated specs.

### CODE REVIEW

**Comment by tpharrison** [comment]
@rmtolmach I'm getting an error - _Require backend-review-group approval_.  Do I have to update the codeowner file and assign the backend-review-group?

FYI - the LoC count is beyond the threshold, but most of the changes are just removing legacy code.

**Comment by rmtolmach** [comment]
> @rmtolmach I'm getting an error - _Require backend-review-group approval_. Do I have to update the codeowner file and assign the backend-review-group to some of these files?
> 
> FYI - the LoC count is beyond the threshold, but most of the changes are just removing legacy code.

@tpharrison that check will fail until you get a review from backend-review-group. Rachal is on support this sprint, so once it's out of draft mode, she'll see it in her queue of PRs. But definitely get a review from a teammate first! :)

**Comment by RachalCassity** [comment]
Please update your branch to fix Audit Service Tags and Code Checks failures

File: spec/support/vcr_cassettes/lighthouse/direct_deposit/update/200_valid.yml
Comment (RachalCassity) [inline spec/support/vcr_cassettes/lighthouse/direct_deposit/update/200_valid.yml:65]: The new standard is to hide the uri domain in vcr. The uri can be defined in the[ vcr configuration](https://github.com/department-of-veterans-affairs/vets-api/blob/16239c47337635c36aa1b7d2cfd474d469513227/spec/support/vcr.rb#L9). 

Ex:
http_interactions:
- request:
    method: put
    uri: <EVSS_BASE_URL>/services/direct-deposit-management/v1/direct-deposit?icn=1012666073V986297

Comment (tpharrison) [reply spec/support/vcr_cassettes/lighthouse/direct_deposit/update/200_valid.yml:65]: @RachalCassity Do you want me to update all of the profile direct deposit cassettes or just the ones changed in this PR?

Comment (tpharrison) [reply spec/support/vcr_cassettes/lighthouse/direct_deposit/update/200_valid.yml:63]: @RachalCassity I updated the cassettes that were modified in this PR and will continue to update cassettes my team owns in another PR.

Comment (tpharrison) [reply spec/support/vcr_cassettes/lighthouse/direct_deposit/update/200_valid.yml:63]: @RachalCassity Who should I list as owners for `config/initializers/integration_recorder.rb`?

**Review by ericboehs — APPROVED** [review]

**Pull request merged**
